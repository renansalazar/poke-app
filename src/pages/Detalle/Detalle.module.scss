.pokedex{
  max-width: 900px;
  margin: auto;
  display: grid;
  grid-template-columns: 1fr 1fr;
  .first{
    background: red;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    border-radius: 50px 0 0 50px;
    padding: 20px;
  }
}


.joystick{
  width: 100%;
  padding: 20px;
  display: flex;
  justify-content: space-between;
}

.pokedex .first {
  .button{
    background: black;
    border-radius: 50%;
    width: 80px;
    height: 80px;
  }
  
  .cross{
    width: 100px;
    height: 100px;
    background: black;
    clip-path: polygon(0 38%, 38% 38%, 38% 0%, 64% 0%, 64% 38%, 100% 38%, 
    100% 64%, 64% 64%, 64% 100%, 38% 100%, 38% 64%, 0 64%);
  }
}


.pokedex .second {
  background-color: red;
  border-radius: 0 0 50px 0;
  .console{
    margin: 20px;
    padding: 20px;
    border-radius: 10px;
    background-color: black;
    color: white;
    display: flex;
    flex-direction: column;
  }
  .containerTypes{
    display: flex;
  }
  
  .stats{
    display: grid;
    grid-auto-columns: 1fr;
    grid-auto-flow: column;
  }
  
  .stats>div{
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: center;
  }
  
  .stats .bar{
    margin-top: 2px;
    width: 20px;
    height: 1px;
    display: block;
    background-color: yellow;
    content: '';
  }
  
  .stats>div .statName{
    font-size: .9rem;
    height: 50px;
  }
}


.containerTriaje{
  display: flex;
  justify-content: space-around;
  .console{
    flex-direction: row!important;
    align-items: center;
    gap: 6px;
    color: yellow!important;
    h5{
      margin: 0;
    }
  }
  
}


.containerName{
  position: absolute;
  color: black;
  bottom: 18px;
  z-index: 9;
  text-transform: capitalize;
  width: 220px;
  font-weight: bold;
  text-align: center;
}

.typeItem{
  margin: auto 6px;
  background: orange;
  border-radius: 6px;
  padding: 6px;
  display: flex;
  align-items: center;
}


.containerSprite{
  position: relative;
  height: 260px;
  background-color: yellow;
  width: 260px;
  padding: 20px;
  margin: 20px;
  border-radius: 16px;
  overflow: hidden;
  img{
    background-color: white;
    position: absolute;
    width: 220px;
    height: 220px;
    margin: auto;
    &:nth-child(1){
      z-index: 1;
      opacity: 1;
      animation: fadeinout 4s infinite backwards;
    }
    @keyframes fadeinout {
      0%,49% { opacity: 0; }
      50%,100% { opacity: 1; }
    }
  }
}


