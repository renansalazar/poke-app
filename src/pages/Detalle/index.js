import {useParams} from "react-router-dom";
import Buscador from "../../components/Buscador";
import { useState } from "react";
import {usePokemon} from "../../hooks/usePokemon";

export default function Detalle () {
  const [photo, ] = useState('default')
  let { name } = useParams()
  const { isLoading, error, data } = usePokemon('repoData', name)

  return (<div>
    {
      error ? <div>No se encontro este pokemon, Â¿no sera un pokemon legendario aun no descubierto? ...</div> : isLoading ? <div>Cargando ...</div> 
      : (<div className='pokedex'>
          <section className='first'>
            {
              photo==='default' && data.data.sprites.front_default &&
              <div className='containerSprite'>
              <img src={data.data.sprites.back_default} alt="back_default" />
              <img src={data.data.sprites.front_default} alt="front_default" />
              <span className='containerName'>{name}</span>
            </div>
            }
            {
              photo==='female' && data.data.sprites.front_female && <div className='containerSprite'>
                <img src={data.data.sprites.back_female} alt="back_female" />
                <img src={data.data.sprites.front_female} alt="front_female" />
              </div>
            }
            {
              photo==='shiny' && data.data.sprites.front_shiny &&
              <div className='containerSprite'>
                <img src={data.data.sprites.back_shiny} alt="back_shiny" />
                <img src={data.data.sprites.front_shiny} alt="front_shiny" />
              </div>
            }
            {
              photo==='shiny_female' && data.data.sprites.front_shiny_female && <div className='containerSprite'>
                <img src={data.data.sprites.back_shiny_female} alt="back_shiny_female" />
                <img src={data.data.sprites.front_shiny_female} alt="front_shiny_female" />
              </div>
            }
            <div className='joystick'>
              <div className='button'>
              </div>
              <div className='cross'>
              </div>
            </div>
              <Buscador name={name} />
          </section>
          <section className='second'>
            <div className='console'>
              <div className="containerTypes">
                <h6>Tipo</h6>
                {
                  data.data.types.map(d=>{
                    return <label className="typeItem">{d.type.name}</label>
                  })
                }
              </div>
              <div className='stats'>
                {
                  data.data.stats.map(s=>{
                    const eleArr = []
                    for(let i = 1; i<=Number(s.base_stat); i++){
                      eleArr.push(<span className="bar"></span>)
                    }
                    return (<div>
                      <small>{s.base_stat}</small>
                      {eleArr}
                      <span className='statName'>{s.stat.name}</span>
                      </div>
                    )
                  })
                }
              </div>
            </div>
            <div className="containerTriaje">
              <div className="console">
                <h5>Altura</h5>
                {
                  Number(data.data.height)/10 + "M"
                }
              </div>
              <div className="console">
                <h5>Peso</h5>
                {
                  Number(data.data.weight)/10+ "KG"
                }
              </div>
            </div>
          </section>
        </div>)
    }
  </div>)
}
