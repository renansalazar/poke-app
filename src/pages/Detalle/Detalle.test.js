import React from 'react';
import {render} from '@testing-library/react';
import Detalle from './';
import { QueryClient, QueryClientProvider } from 'react-query';

describe('Detalle component', () => {
  test('first call to detail', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Detalle />
      </QueryClientProvider>)
    
    expect(container.firstChild).toMatchSnapshot()
  });
});