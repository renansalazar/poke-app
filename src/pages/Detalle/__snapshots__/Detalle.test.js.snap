// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detalle component first call to detail 1`] = `
<div>
  <div>
    Cargando ...
  </div>
</div>
`;
